core/src/main/java/nl/nn/adapterframework/pipes/JwtPipe.java

```diff
@@ -0,0 +1,168 @@
+			} catch (KeyLengthException e) {
+				throw new ConfigurationException("invalid shared key", e);
+			}
+		}
+
+		if(globalSigner == null && getParameterList().findParameter(SHARED_SECRET_PARAMETER_NAME) == null) {
+			throw new ConfigurationException("must either provide a [sharedSecret] (alias) or parameter");
+		}
+	}
+
+	@Override
+	public PipeRunResult doPipe(Message message, PipeLineSession session) throws PipeRunException {
+		final JWSSigner signer;
+		Builder claimsSetBuilder = new JWTClaimsSet.Builder();
+
+		Map<String, Object> parameterMap = getParameterValueMap(message, session);
+		if(parameterMap != null) {
+			Object sharedKey = parameterMap.remove(SHARED_SECRET_PARAMETER_NAME);
+			parameterMap.forEach(claimsSetBuilder::claim);
+
+			signer = getSigner(sharedKey);
+		} else {
+			signer = globalSigner;
+		}
+
+		if(expirationTime > 0) {
+			Date expirationDate = Date.from(Instant.now().plusSeconds(expirationTime));
+			claimsSetBuilder.expirationTime(expirationDate);
+		}
+		claimsSetBuilder.issueTime(Date.from(Instant.now()));
+
+		String jwtToken = createAndSignJwtToken(signer, claimsSetBuilder.build());
+		return new PipeRunResult(getSuccessForward(), Message.asMessage(jwtToken));
+	}
+
+	private JWSSigner getSigner(Object sharedKey) throws PipeRunException {
+		if(Objects.nonNull(sharedKey)) {
+			try {
+				return new MACSigner(Message.asString(sharedKey));
+			} catch (KeyLengthException | IOException e) {
+				throw new PipeRunException(this, "invalid shared key", e);
+			}
+		}
+		return globalSigner;
+	}
+
+	private Map<String, Object> getParameterValueMap(Message message, PipeLineSession session) throws PipeRunException {
+		ParameterList parameterList = getParameterList();
+		if(parameterList != null) {
+			ParameterValueList pvl;
+			try {
+				pvl = parameterList.getValues(message, session);
+			} catch (ParameterException e) {
+				throw new PipeRunException(this, "unable to resolve parameters", e);
+			}
+			return pvl.getValueMap();
+		}
+		return null;
+	}
+
+	private @Nonnull String createAndSignJwtToken(@Nonnull JWSSigner signer, @Nonnull JWTClaimsSet claims) throws PipeRunException {
+		SignedJWT signedJWT = new SignedJWT(jwtHeader, claims);
+
+		try {
+			signedJWT.sign(signer);
+		} catch (JOSEException e) {
+			throw new PipeRunException(this, "unable to sing JWT using [" + signer + "]", e);
+		}
+
+		String jwt = signedJWT.serialize();
+		log.debug("generated JWT token [{}]", jwt);
+		return jwt;
+	}
+
+	/** Authentication alias used for the SharedSecret */
+	public void setAuthAlias(String alias) {
+		this.sharedSecretAlias = alias;
+	}
+	/** Authentication alias used for authentication to the host */
+	public void setSharedSecret(String sharedSecret) {
+		this.sharedSecret = sharedSecret;
+	}
+
+	/**
+	 * ExpirationTime in seconds, 0 to disable
+	 * @ff.default 600
+	 */
+	public void setExpirationTime(int expirationTime) {
+		this.expirationTime = expirationTime;
+	}
```

Comment:
I thought that FrankDoclet also supports Lombok and Javadoc. So you can add `@Setter` at the var declaration and add Javadoc there too. Isn't that possible?

Reply:
I believe it does, though it's not implemented like this (yet).
It's also personal preference I suppose, I'm not too fond of all the javadoc at the top of each class.